File: boot-basic/src/main/java/com/lucas/bootbasic/config/AsyncConfig.java
package com.lucas.bootbasic.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;

import java.util.concurrent.Executor;

/**
 * @package : com.lucas.bootbasic.config
 * @name : AsyncConfig.java
 * @date : 2025. 9. 26. 오후 3:44
 * @author : lucaskang(swings134man)
 * @Description: 
**/
@Configuration
@EnableAsync
public class AsyncConfig {

    @Bean("taskExecutor")
    public Executor taskExecutor() {
        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
        executor.setCorePoolSize(5);
        executor.setMaxPoolSize(10);
        executor.setQueueCapacity(25);
        executor.setThreadNamePrefix("Async-");
        executor.initialize();
        return executor;
    }
}

File: hexagonal-kotlin/src/main/kotlin/com/lucas/hexagonalkotlin/infrastructure/common/AsyncConfig.kt
package com.lucas.hexagonalkotlin.infrastructure.common

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableAsync
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor
import java.util.concurrent.Executor
import java.util.concurrent.ThreadPoolExecutor

/**
 * AsyncConfig.kt: Async Config Class
 *
 * @author: lucaskang(swings134man)
 * @since: 2025. 9. 26. 오전 2:28
 * @description: 
 */
@EnableAsync
@Configuration
class AsyncConfig {

    @Bean("multiExecutor")
    fun multiExecutor(): Executor {
        val executor = ThreadPoolTaskExecutor()
        executor.corePoolSize = 5
        executor.maxPoolSize = 10
        executor.queueCapacity = 500 // 대기 queue size
        executor.setThreadNamePrefix("AsyncMulti-")
        executor.initialize()
        return executor
    }
}